src/Workspaces/Remote/ServiceHub/Host/SolutionAssetCache.cs

```diff
@@ -63,11 +69,13 @@ public SolutionAssetCache(TimeSpan cleanupInterval, TimeSpan purgeAfter, TimeSpa
             Task.Run(CleanAssetsAsync, CancellationToken.None);
         }
 
-        public bool TryAddAsset(Checksum checksum, object value)
+        public object GetOrAdd(Checksum checksum, object value)
```

Comment:
changing from TryAdd to GetOrAdd pattern, this way, if two callers compute the same checksum-value pair, only one wins.